JUnit5-README.md

```diff
@@ -0,0 +1,193 @@
+
+```
+
+### Add maven dependencies
+
+```xml
+<dependencies>
+  <dependency>
+    <groupId>org.wildfly.extras.sunstone</groupId>
+    <artifactId>sunstone-junit5</artifactId>
+    <version>${version.org.wildfly.extras.sunstone}</version>
+  </dependency>
+</dependencies>
+```
+WildFly client libraries must also be specified as described
+in [Creaper library README](https://github.com/wildfly-extras/creaper/blob/master/README.md#jboss-as-7--wildfly-client-libraries).
+
+### Use Sunstone extension
+To enable the extension, simply add annotation on a test class:
+```Java
+@Sunstone
+public class Test {}
+```
+
+
+
+### Describe cloud deployment
+You can describe resources to be deployed using annotations `@With*` on a test class. See following examples for Azure/AWS specific resources or JClouds that provide abstraction for several Cloud resources.
+
+#### Azure templates
+
+```java
+@Sunstone
+@WithAzureArmTemplate("appservices/clusteredWebApp.json")
+public class AzureTest {}
+```
+
+In this case, template is located in resources. You can also provide a raw content or URL and the template shall be downloaded. Resources are deployed to the group and region defined in `sunstone.properties`.
+
+You can download JSON definition of already existing resources or downloading JSON for automation in create form.
+
+Azure template allows you to deploy and configure Azure specific resources using native technologies.
+
+#### AWS CloudFormation template
+
+```java
+@Sunstone
+@WithAwsCfTemplate("aws/messaging-mdb.yaml")
+public class AWSTest {}
+```
+
+In this case, template is located in resources. You can also provide a raw content or URL and the template shall be downloaded. Resources are deployed to a stack of random name in region defined in `sunstone.properties`
+
+You can use Former2 tool to obtain YAML definition of already existing resources.
+
+CloudFormation template allows you to deploy and configure AWS specific resources using native technologies.
+
+#### JClouds
+TODO
+
+```java
+@Sunstone
+@WithJClouds("test.properties")
+public class JCloudTest {}
+```
+
+In this case, property file is located in resources. You can also provide a raw content or URL and the file shall be downloaded.
+
+Contract for different providers can be found at:
+- [azure-arm-README](azure-arm-README.md)
+- [baremetal-README](baremetal-README.md)
+- [ec2-README](ec2-README.md)
+- [docker-README](docker-README.md)
+- [openstack-README](openstack-README.md)
+- [wildfly-README](wildfly-README.md)
+
+JCloud enables you to share a test among different providers.
+
+### Provide additional configuration
+Sunstone allows you to define a setup task with enabled DI. Add the `@Setup` annotation on test class
+
+```java
+@Sunstone
+@WithAwsCfTemplate("aws/messaging-mdb.yaml")
+@Setup(Configure2LCApp.class)
+public class AWSTest {}
+```
+```java
+
+public class Configure2LCApp extends AbstractSetupTask {
+    
+  @SunstoneResource(of = "eapqe-instance", hint = AWS_EC2_INSTANCE)
```

Comment:
OK, this is interesting and looks like a powerful concept, but it also confuses me. Where's `eapqe-instance` defined? In `aws/messaging-mdb.yaml`? Where can I see the possible values for `hint`? And why is something with a hint of `AWS_EC2_INSTANCE` being injected into an object of `OnlineManagementClient`?

Reply:
`hint` is the enum `SunstoneResourceHint`. IDE will help you. The concept is, I want `OnlineManagementClient` for `eapqe-instance` that is defined deployed by `aws/messaging-mdb.yaml` (contains EC2 instance with the same name). Hint is needed because resources may have same name in AWS.